src/hazelcore/dynamics/Elaborator_Exp.re

```diff
@@ -93,7 +93,12 @@ and syn_elab_line =
               ),
             )
           };
-        let d1 = DHExp.cast(d1, ty1', ty1);
+        let d1 =
+          switch (ty1) {
```

Comment:
this logic might be better to put into DHExp.cast function since I think you'll need it in a few places -- a cast with target type Unknown(SynPatternVar) should be a no op